src/jbi/bugzilla_objects.py

```diff
@@ -0,0 +1,245 @@
+"""
+Bugzilla Typed Objects for ease of use throughout JBI
+"""
+import datetime
+from typing import Dict, List, Optional, Tuple
+from urllib.parse import ParseResult, urlparse
+
+from pydantic import BaseModel  # pylint: disable=no-name-in-module
+
+
+class BugzillaWebhookUser(BaseModel):
+    """Bugzilla User Object"""
+
+    id: int
+    login: str
+    real_name: str
+
+
+class BugzillaWebhookEventChange(BaseModel):
+    """Bugzilla Change Object"""
+
+    field: str
+    removed: str
+    added: str
+
+
+class BugzillaWebhookEvent(BaseModel):
+    """Bugzilla Event Object"""
+
+    action: str
+    time: Optional[datetime.datetime] = None
+    user: Optional[BugzillaWebhookUser] = None
+    changes: Optional[List[BugzillaWebhookEventChange]] = None
+    target: Optional[str] = None
+    routing_key: Optional[str] = None
+
+
+class BugzillaWebhookAttachment(BaseModel):
+    """Bugzilla Attachment Object"""
+
+    content_type: Optional[str]
+    creation_time: Optional[datetime.datetime]
+    description: Optional[str]
+    file_name: Optional[str]
+    flags: Optional[List]
+    id: int
+    is_obsolete: Optional[bool]
+    is_patch: Optional[bool]
+    is_private: Optional[bool]
+    last_change_time: Optional[datetime.datetime]
+
+
+class BugzillaWebhookComment(BaseModel):
+    """Bugzilla Comment Object"""
+
+    body: Optional[str]
+    id: Optional[int]
+    number: Optional[int]
+    is_private: Optional[bool]
+    creation_time: Optional[datetime.datetime]
+
+    def is_comment_description(self) -> bool:
+        """Used to determine if `self` is a description or comment."""
+        if self.number in [0]:  # Currently 0th comment is the description
+            return True
+        return False
+
+    def is_comment_generic(self) -> bool:
+        """All comments after comment-0 are generic"""
+        is_description = self.is_comment_description()
+        return not is_description
+
+    def is_private_comment(self) -> bool:
+        """Helper function to determine if this comment private--not accessible or open"""
+        if self.is_private is None:
+            self.is_private = False
+        return self.is_private
+
+
+class BugzillaBug(BaseModel):
+    """Bugzilla Bug Object"""
```

Comment:
Is there Bugzilla documentation we can link to here? Might be a useful reference in the future.